### DISCUSSION TITLE
Modal freeze with multiple usage at the same time

### DISCUSSION THREAD

**Discussion by DARL0K** [discord-php/DiscordPHP#1118] [Q&A] [answered]
Hi. I have trouble when couple of users at the same time using modals. For example this code:
```
$discord->listenCommand("testreg", function (Interaction $interaction) use ($discord) {
    $ar[] = ActionRow::new()->addComponent(TextInput::new("Test teest:",TextInput::STYLE_SHORT)->setMinLength(1));

    $interaction->showModal("test", "123123", $ar, function (Interaction $interaction, Collection $components) {
        $interaction->acknowledge();
    });
});
```
Explanation of the issue:
User 1 wrote /testreg and filling the form. At the same time User 2 wrote /testreg and also started to fill the form.
User 1 or User 2 will recieve "Something went wrong. Try again.". But only one of them, another will have successfull complete.

![image](https://github.com/discord-php/DiscordPHP/assets/36866683/c2e119b2-b1e8-4984-9d38-b327a211896c)

May be someone could clarify in which way I should think?

**Comment by key2peace** [comment]
The second arg of showModal is a custom_id, which should be unique per interaction

**Reply by DARL0K** [reply]
Hi. I've already tested that -> same result.

```
include __DIR__ . "/vendor/autoload.php";

use Discord\Discord;
use Discord\Helpers\Collection;
use Discord\Parts\Guild\Guild;
use Discord\Parts\Guild\Role;
use Discord\Parts\Channel\Channel;
use Discord\Parts\Channel\Message;
use Discord\Parts\Embed\Embed;
use Discord\Parts\User\Activity;
use Discord\WebSockets\Intents;
use Discord\WebSockets\Event;
use Discord\Builders\MessageBuilder;
use Discord\Builders\Components\SelectMenu;
use Discord\Builders\Components\TextInput;
use Discord\Builders\Components\ActionRow;
use Discord\Builders\Components\Option;
use Discord\Builders\Components\Button;
use Discord\Builders\Components\Component;
use Discord\Builders\CommandBuilder;
use Discord\Parts\Interactions\Command\Command;
use Discord\Parts\Interactions\Command\Option as Options;
use Discord\Parts\Interactions\Interaction;
use Discord\Parts\WebSockets\VoiceStateUpdate;
use React\EventLoop\Loop;

$discord = new Discord([
    "token" => "",
    "intents" =>
        Intents::getDefaultIntents() |
        Intents::GUILD_MEMBERS |
        Intents::GUILD_VOICE_STATES |
        Intents::GUILD_INVITES, // default intents as well as guild members
    "loadAllMembers" => true,
    "storeMessages" => true,
    "retrieveBans" => false,
    "loop" => Loop::get(),
]);

$discord->on("ready", function (Discord $discord) {
	
	$builder = CommandBuilder::new()
        ->setName("testreg")
        ->setDescription("test");
	$command = new Command($discord, $builder->toArray());
	
	
	$discord->listenCommand("testreg", function (Interaction $interaction) use ($discord) {
		$ar[] = ActionRow::new()->addComponent(TextInput::new("Test teest:",TextInput::STYLE_SHORT)->setMinLength(1));
		
		$customID = rand(1, 1000000);
		echo "Custom id: ".$customID.PHP_EOL;
		
		$interaction->showModal("test", $customID, $ar, function (Interaction $interaction, Collection $components) {
			$interaction->acknowledge();
		});
    });
	
	
});
```

**Comment by DARL0K** [comment]
Hi Guys. Looks like this thing is not usable. Is it a bug in Discord or DiscordPHP lib?

**Comment by SQKo** [comment]
> The second arg of showModal is a custom_id, which should be unique per interaction

I think it is not the problem

I'm not able to reproduce right now, but I think i know where is the mistake, so yea it's probably a bug.

**Comment by SQKo** [comment, marked as answer]
> Hi Guys. Looks like this thing is not usable. Is it a bug in Discord or DiscordPHP lib?

Should be fixed in dev-master#[d5ec759](https://github.com/discord-php/DiscordPHP/commit/d5ec759f32c7a4118a869dd7e8d90be5217db2ed)

**Comment by DARL0K** [comment]
> Should be fixed in dev-master#[d5ec759](https://github.com/discord-php/DiscordPHP/commit/d5ec759f32c7a4118a869dd7e8d90be5217db2ed)

I'll test it soon. Thank you

**Comment by DARL0K** [comment]
> > Hi Guys. Looks like this thing is not usable. Is it a bug in Discord or DiscordPHP lib?
> 
> Should be fixed in dev-master#[d5ec759](https://github.com/discord-php/DiscordPHP/commit/d5ec759f32c7a4118a869dd7e8d90be5217db2ed)

I'm sorry, forgot to update this ticket. Issue is gone. Thank you. Now it works as it should
